### PR TITLE
Modify upstream info to update correctly

### PR DESCRIPTION
**Pull request by Gabib32**
The upstream info struct should only contain information
appropriate to the command being called. Therefore, create
an instance of the struct within the command call, and have
specific fields updated if they need to be, or are empty.

Signed-off-by: gabrielle beyer <gabrielle.n.beyer@intel.com>

### CODE REVIEW

File: cmd/fetch.go [hunk 1]
```diff
@@ -96,28 +97,35 @@ func init() {
...
 	fetchAllCmd.Flags().StringVarP(&fetchFlags.bundleURL, "bundleurl", "b", "", "URL from which to pull bundle definitions")
 	fetchAllCmd.Flags().StringVarP(&fetchFlags.upstreamURL, "upstreamurl", "u", "", "URL from which to pull update metadata")
 	fetchAllCmd.Flags().BoolVar(&fetchFlags.update, "update", false, "update pre-existing Repo")
+	fetchAllCmd.Flags().BoolVarP(&fetchFlags.recursive, "recursive", "r", false, "recursively fetch all content referenced in update metadata")
 
 	fetchBundlesCmd.Flags().StringVarP(&fetchFlags.bundleURL, "bundleurl", "b", "", "URL from which to pull bundle definitions")
 
 	fetchRepoCmd.Flags().StringVarP(&fetchFlags.version, "version", "v", "", "version from which to pull data")
+	fetchRepoCmd.Flags().StringVarP(&fetchFlags.upstreamURL, "upstreamurl", "u", "", "URL from which to pull update metadata")
+	fetchRepoCmd.Flags().BoolVar(&fetchFlags.update, "update", false, "update pre-existing Repo")
```
Comment (matthewrsj) [inline cmd/fetch.go]: `Repo` does not need to be capitalized here.

File: internal/helpers/helpers.go [hunk 2]
```diff
@@ -296,6 +296,15 @@ func FailIfErr(err error) {
 	}
 }
 
+// GetVersionAsUint converts a version string into a uint
+func GetVersionAsUint(ver string) (uint, error) {
```
Comment (matthewrsj) [inline internal/helpers/helpers.go]: I don't see the point of this function. It is just wrapping a single call to the std library `strconv.Atoi`, handling the error, and returning the cast value.

If you still want to keep the function, call it something more generic. There is nothing preventing this from being called on any numeric string but the name implies you are `Get`ting a `Version` when all you are doing is converting a string.

A better name for this function would be `ParseUint`...

File: cmd/fetch.go (hunk 1 above)
Comment (Gabib32) [reply cmd/fetch.go]: :pick: e

Comment (Gabib32) [reply cmd/fetch.go]: :stuck_out_tongue_winking_eye:

File: internal/helpers/helpers.go (hunk 2 above)
Comment (tmarcu) [reply internal/helpers/helpers.go]: https://golang.org/pkg/strconv/#ParseUint

Comment (Gabib32) [reply internal/helpers/helpers.go]: Yikes, yeah I have no idea what I was thinking with this.. /me fixing

Comment (Gabib32) [reply internal/helpers/helpers.go]: :( ParseUint is uint64, so still needs the masking to uint

**Pull request closed without merging**